commit cf8aa9bf97cadf85745506c6a3e244b22c268d63
Author: Kees Cook <kees@kernel.org>
Date:   Sat Sep 24 00:33:15 2022 -0700

    ovl: Use "buf" flexible array for memcpy() destination
    
    The "buf" flexible array needs to be the memcpy() destination to avoid
    false positive run-time warning from the recent FORTIFY_SOURCE
    hardening:
    
      memcpy: detected field-spanning write (size 93) of single field "&fh->fb"
      at fs/overlayfs/export.c:799 (size 21)
    
    Reported-by: syzbot+9d14351a171d0d1c7955@syzkaller.appspotmail.com
    Link: https://lore.kernel.org/all/000000000000763a6c05e95a5985@google.com/
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Reviewed-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>

diff --git a/fs/overlayfs/overlayfs.h b/fs/overlayfs/overlayfs.h
index eee8f08d32b6..e74a610a117e 100644
--- a/fs/overlayfs/overlayfs.h
+++ b/fs/overlayfs/overlayfs.h
@@ -108,7 +108,7 @@ struct ovl_fh {
 	u8 padding[3];	/* make sure fb.fid is 32bit aligned */
 	union {
 		struct ovl_fb fb;
-		u8 buf[0];
+		DECLARE_FLEX_ARRAY(u8, buf);
 	};
 } __packed;